### PR TITLE
Added `zstd-static` package to the conda environment configuration files.

### PR DESCRIPTION
**Pull request by kmr-srbh**
Also incorporated changes to installation-docs suggested in #2537 .

### CODE REVIEW

File: environment_unix.yml [hunk 1]
```diff
@@ -2,16 +2,18 @@ name: lp
...
+  - git
   - pip
-  - python=3.10.2
+  - make
   - re2c
-  - setuptools
   - toml
   - zlib
-  - git
+  - flex
```
Comment (certik) [inline environment_unix.yml:11]: Why is flex needed? I don't think it's needed.

Comment (kmr-srbh) [reply environment_unix.yml:11]: Oh. I am sorry for the mistake. I recall installing `flex` when I ran into an error with `bison`. I recalled [this](https://stackoverflow.com/questions/7320370/installation-of-flex-and-bison-in-ubuntu) thread.

I am making the amends.

**Comment by kmr-srbh** [comment]
@certik Could you please investigate as to why one of the above tests failed? I checked the log and the workflow file. Nothing seems to explain the reason for only 1 test failing for `Python 3.11`.

**Review by certik — APPROVED** [review]
I think that's fine.

**Comment by certik** [comment]
The failure is due to https://github.com/lcompilers/lpython/issues/2542.

File: environment_unix.yml [hunk 2]
```diff
@@ -2,16 +2,17 @@ name: lp
...
   - re2c
-  - setuptools
   - toml
   - zlib
-  - git
+  - cmake
+  - numpy
+  - flake8
+  - setuptools
+  - bison=3.4
```
Comment (Shaikh-Ubaid) [inline environment_unix.yml:15]: Is it needed to change the order of specifying dependencies here? It seems hard to interpret which packages where removed and which ones were added.

File: README.md [hunk 3]
```diff
@@ -36,16 +36,10 @@ Follow the steps below to install and run LPython on Linux, Windows or macOS.
 
 - ### Set up your system
     - Linux
-        - Make sure you have `g++` and `cmake` installed. If not, install them using the below command:
+        - Run the following command to install some global build dependencies:
```
Comment (Shaikh-Ubaid) [inline README.md:39]: I think the first step in linux should be creating the conda environment. The next step can be installing the required global build dependencies.

Similarly, update for `docs/src/installation.md`.

File: environment_win.yml [hunk 4]
```diff
@@ -2,18 +2,19 @@ name: lp
...
   - pip
+  - make
+  - re2c
   - toml
+  - zlib
+  - cmake
+  - ninja
+  - numpy
+  - xonsh
+  - flake8
```
[the remark this answers is not in the document]
Comment (kmr-srbh) [reply environment_win.yml:15]: In both the files, only the `zstd-static` package was added.

File: README.md (hunk 3 above)
Comment (kmr-srbh) [reply README.md:39]: I understand your point @Shaikh-Ubaid, but moving this section will logically separate the conda environment creation part for Linux and Windows. As installing the global dependencies takes some time for a new contributor, I let it remain at the top. 

The user can perhaps download Miniforge in the meantime. 😄

Comment (Shaikh-Ubaid) [reply README.md:39]: Ok, it seems fine for now.

File: environment_win.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shaikh-Ubaid) [reply environment_win.yml:15]: Ok, good.

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good to me. Thanks for this!

**Pull request merged**